include/seastar/util/process.hh

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2022 Kefu Chai ( tchaikov@gmail.com )
+ */
+
+#pragma once
+
+#include <sys/types.h>
+#include <string_view>
+#include <variant>
+#include <vector>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/sstring.hh>
+
+namespace seastar::experimental {
+
+/// Interact with a spawned subprocess
+///
+/// \note the spawned subprocess should always be \c wait()'ed. Otherwise,
+/// the Seastar application spawning the subprocess will leave us with
+/// one ore more zombie subprocesses after it exists.
+class process {
+    struct create_tag {};
+public:
+    process(create_tag, pid_t pid, int stdin, int stdout, int stderr);
```

Comment:
The constructor can be made private, it's only called from places that have access to create_tag

Reply:
no. i don't think so. it's the lambda which creates calls the ctor. and it'd be difficult to mark the lambda `process::spawn()` overloads `friend` without using an proxy method, which can be marked as a friend though.